lib/active_merchant/billing/gateways/airwallex.rb

```diff
@@ -0,0 +1,257 @@
+      self.supported_countries = %w[AT AU BE BG CY CZ DE DK EE GR ES FI FR GB HK HR HU IE IT LT LU LV MT NL PL PT RO SE SG SI SK]
+      self.default_currency = 'AUD'
+      self.supported_cardtypes = %i[visa master]
+
+      self.homepage_url = 'https://airwallex.com/'
+      self.display_name = 'Airwallex'
+
+      ENDPOINTS = {
+        login: '/authentication/login',
+        setup: '/pa/payment_intents/create',
+        sale: '/pa/payment_intents/%{id}/confirm',
+        capture: '/pa/payment_intents/%{id}/capture',
+        refund: '/pa/refunds/create',
+        void: '/pa/payment_intents/%{id}/cancel'
+      }
+
+      def initialize(options = {})
+        requires!(options, :client_id, :client_api_key)
+        @client_id = options[:client_id]
+        @client_api_key = options[:client_api_key]
+        super
+        setup_ids(options) unless options[:request_id] && options[:merchant_order_id]
+        setup_access_token
+      end
+
+      def purchase(money, card, options = {})
+        payment_intent = create_payment_intent(money, @options)
+        post = {
+          'request_id' => update_request_id(@options, 'purchase'),
+          'return_url' => options[:return_url]
+        }
+        add_card(post, card, options)
+        add_payment_method_options(post, card, options)
+
+        commit(:sale, post, payment_intent)
+      end
+
+      def authorize(money, payment, options = {})
+        # authorize is just a purchase w/o an auto capture
+        purchase(money, payment, options.merge({ auto_capture: false }))
+      end
+
+      def capture(money, authorization, options = {})
+        raise ArgumentError, 'An authorization value must be provided.' if authorization.blank?
+
+        update_request_id(@options, 'capture')
+        post = {
+          'request_id' => @options[:request_id],
+          'amount' => amount(money)
+        }
+        commit(:capture, post, authorization)
+      end
+
+      def refund(money, authorization, options = {})
+        raise ArgumentError, 'An authorization value must be provided.' if authorization.blank?
+
+        post = {}
+        post[:amount] = amount(money)
+        post[:payment_intent_id] = authorization
+        post[:request_id] = update_request_id(@options, 'refund')
+        post[:merchant_order_id] = @options[:merchant_order_id]
+
+        commit(:refund, post)
+      end
+
+      def void(authorization, options = {})
+        raise ArgumentError, 'An authorization value must be provided.' if authorization.blank?
+
+        update_request_id(@options, 'void')
+        post = {}
+        post[:request_id] = @options[:request_id]
+        commit(:void, post, authorization)
+      end
+
+      # def credit(money, credit_card, options = {})
+      # end
+
+      def verify(credit_card, options = {})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
```

Comment:
Did anyone reach out to Airwallex to confirm that $0 auths are not supported?

Reply:
@dsmcclain I don't remember if we did, but I do think I tried it and it did not work. It might be a thing that's settable at the account level, but I think it makes sense for that to be a thing we save for future development.

Reply:
Ok, saving for future development seems totally reasonable to me. It will for sure be something customers ask for down the line.